applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -238,7 +238,17 @@ def AdvanceInTime(self, current_time):
         return new_time
 
     def ComputeDeltaTime(self):
-        return self.settings["time_stepping"]["time_step"].GetDouble()
+        if self.settings["time_stepping"].Has("time_step"):
+            return self.settings["time_stepping"]["time_step"].GetDouble()
+        elif self.settings["time_stepping"].Has("time_step_table"):
+            current_time = self.main_model_part.ProcessInfo[KratosMultiphysics.TIME]
+            time_step_table = self.settings["time_stepping"]["time_step_table"].GetMatrix()
+            for interval in range(time_step_table.Size1()-1):
+                if current_time > time_step_table[interval, 0] and current_time <= time_step_table[interval+1, 0]:
+                    return time_step_table[interval, 1]
+            return time_step_table[time_step_table.Size1()-1, 1]
```

Comment:
this does interpolate the time-step, which was the main change requested
I recommend using the table from the core, with which it should be really straight forward, since it has a method for interpolation

Reply:
OK, I have not understand that

Reply:
Done